aws/resource_aws_default_security_group_test.go

```diff
@@ -12,66 +12,167 @@ import (
+					resource.TestCheckResourceAttr(resourceName, "egress.#", "0"),
+				),
+			},
+			{
+				ResourceName:            resourceName,
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"revoke_rules_on_delete"},
+			},
+		},
+	})
+}
+
+func TestAccAWSDefaultSecurityGroup_Classic_basic(t *testing.T) {
+	oldvar := os.Getenv("AWS_DEFAULT_REGION")
+	os.Setenv("AWS_DEFAULT_REGION", "us-east-1")
+	defer os.Setenv("AWS_DEFAULT_REGION", oldvar)
+
+	var group ec2.SecurityGroup
+	resourceName := "aws_default_security_group.test"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:      func() { testAccPreCheck(t); testAccEC2ClassicPreCheck(t) },
+		IDRefreshName: resourceName,
 		Providers:     testAccProviders,
 		CheckDestroy:  testAccCheckAWSDefaultSecurityGroupDestroy,
 		Steps: []resource.TestStep{
 			{
-				Config: testAccAWSDefaultSecurityGroupConfig_classic,
+				Config: testAccAWSDefaultSecurityGroupConfig_Classic,
 				Check: resource.ComposeTestCheckFunc(
-					testAccCheckAWSDefaultSecurityGroupExists("aws_default_security_group.web", &group),
-					testAccCheckAWSDefaultSecurityGroupAttributes(&group),
-					resource.TestCheckResourceAttr(
-						"aws_default_security_group.web", "name", "default"),
-					tfawsresource.TestCheckTypeSetElemNestedAttrs("aws_default_security_group.web", "ingress.*", map[string]string{
+					testAccCheckAWSDefaultSecurityGroupExists(resourceName, &group),
+					resource.TestCheckResourceAttr(resourceName, "name", "default"),
+					resource.TestCheckResourceAttr(resourceName, "description", "default group"),
+					resource.TestCheckResourceAttr(resourceName, "vpc_id", ""),
+					resource.TestCheckResourceAttr(resourceName, "ingress.#", "1"),
+					tfawsresource.TestCheckTypeSetElemNestedAttrs(resourceName, "ingress.*", map[string]string{
 						"protocol":      "tcp",
 						"from_port":     "80",
 						"to_port":       "8000",
 						"cidr_blocks.#": "1",
 						"cidr_blocks.0": "10.0.0.0/8",
 					}),
+					resource.TestCheckResourceAttr(resourceName, "egress.#", "0"),
+					testAccCheckAWSDefaultSecurityGroupARN(resourceName, &group),
+					testAccCheckAWSDefaultSecurityGroupOwnerID(resourceName),
+					resource.TestCheckResourceAttr(resourceName, "tags.%", "1"),
+					resource.TestCheckResourceAttr(resourceName, "tags.Name", "tf-acc-test"),
 				),
 			},
+			{
+				Config:   testAccAWSDefaultSecurityGroupConfig_Classic,
+				PlanOnly: true,
+			},
+			{
+				ResourceName:            resourceName,
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"revoke_rules_on_delete"},
+			},
 		},
 	})
 }
 
+// This test currently fails because it does not clear tags when adopting the resource, resulting
+// in a non-empty plan.
+// https://github.com/terraform-providers/terraform-provider-aws/issues/14631
+// func TestAccAWSDefaultSecurityGroup_Classic_empty(t *testing.T) {
```

Comment:
Instead of commenting this out, it can be implemented and the comment moved to `TestAccSkip("...")` as the first line. 👍